omnigibson/utils/usd_utils.py

```diff
@@ -228,6 +228,139 @@ def create_joint(prim_path, joint_type, body0=None, body1=None, enabled=True,
+                    i += 1
+
+        # Generate rigid body view, making sure to update the simulation first (without physics) so that the physx
+        # backend is synchronized with any newly added objects
+        # We also suppress the omni tensor plugin from giving warnings we expect
+        og.sim.pi.update_simulation(elapsedStep=0, currentTime=og.sim.current_time)
+        with suppress_omni_log(channels=["omni.physx.tensors.plugin"]):
+            cls._CONTACT_VIEW = og.sim.physics_sim_view.create_rigid_contact_view(
+                pattern="/World/*/*",
+                filter_patterns=list(cls._PATH_TO_IDX.keys()),
+            )
+
+        # Sanity check generated view -- this should generate square matrices of shape (N, N, 3)
+        n_bodies = len(cls._PATH_TO_IDX)
+        # from IPython import embed; embed()
+        assert cls._CONTACT_VIEW.sensor_count == n_bodies and cls._CONTACT_VIEW.filter_count == n_bodies, \
+            f"Got unexpected contact view shape. Expected: ({n_bodies}, {n_bodies}); " \
+            f"got: ({cls._CONTACT_VIEW.sensor_count}, {cls._CONTACT_VIEW.filter_count})"
+
+    @classmethod
+    def get_body_idx(cls, prim_path):
+        """
+        Returns:
+            int: idx assigned to the rigid body defined by @prim_path
+        """
+        return cls._PATH_TO_IDX[prim_path]
+
+    @classmethod
+    def get_all_impulses(cls):
+        """
+        Grab all impulses at the current timestep
+
+        Returns:
+            n-array: (N, N, 3) impulse array defining current impulses between all N contact-sensor enabled rigid bodies
+                in the simulator
+        """
+        # Generate the contact matrix if it doesn't already exist
+        if cls._CONTACT_MATRIX is None:
+            cls._CONTACT_MATRIX = cls._CONTACT_VIEW.get_contact_force_matrix(dt=1.0)
+            cls._CONTACT_CACHE = dict()
+
+        return cls._CONTACT_MATRIX
+
+    @classmethod
+    def get_impulses(cls, prim_paths_a, prim_paths_b):
+        """
+        Grabs the matrix representing all impulse forces between rigid prims from @prim_paths_a and
+        rigid prims from @prim_paths_b
+
+        Args:
+            prim_paths_a (list of str): Rigid body prim path(s) with which to grab contact impulses against
+                any of the rigid body prim path(s) defined by @prim_paths_b
+            prim_paths_b (list of str): Rigid body prim path(s) with which to grab contact impulses against
+                any of the rigid body prim path(s) defined by @prim_paths_a
+
+        Returns:
+            n-array: (N, M, 3) impulse array defining current impulses between N bodies from @prim_paths_a and M bodies
+                from @prim_paths_b
+        """
+        # Compute subset of matrix and return
+        idxs_a = [cls._PATH_TO_IDX[path] for path in prim_paths_a]
+        idxs_b = [cls._PATH_TO_IDX[path] for path in prim_paths_b]
+        return cls.get_all_impulses()[idxs_a][:, idxs_b]
```

Comment:
nit: would this work? `cls.get_all_impulses()[idxs_a, idxs_b]`

Also, why is the shape [N, M, 3]? What is the `3`?

Reply:
3 - impulse vector. no contact = 0, 0, 0